### ISSUE TITLE
List of incompatible apps

### ISSUE THREAD

**Issue by Leleat** [Leleat/Tiling-Assist#61]
This is a list of apps, which you expect to be 'tile-able' but aren't for whatever reason...

- **mpv:** it doesn't properly resize. See #40. (It does however work on Wayland.)

**Comment by laichiaheng** [comment]
MPV does resize on Xorg.

**Comment by Leleat** [comment]
> MPV does resize properly on Xorg.
We need to add this in mpv.conf:
no-keepaspect-window

Thanks, updated the post.

**Comment by gilesfun** [comment]
>Maximizing by double-clicking the titlebar (or using a maximize button) can't be handled by this extension, because that happens in mutter and not GNOME Shell. So everything related to it, like for instance gaps on maximized windows, doesn't work

[gnome-shell-extensions-useless-gaps](https://github.com/mipmip/gnome-shell-extensions-useless-gaps) is able to work around this first limitation (and seems to function nicely alongside this extension)

Maybe their implementation could be added to this extension?

**Comment by Leleat** [comment]
> [gnome-shell-extensions-useless-gaps](https://github.com/mipmip/gnome-shell-extensions-useless-gaps) is able to work around this first limitation (and seems to function nicely alongside this extension)
>
>Maybe their implementation could be added to this extension?

Thanks for the suggestion. I've looked at that extension and it looks like it has the same issues I've encountered when trying to handle the 'double click the titlebar' / maximize-button case. Someone else also already submitted a PR (https://github.com/Leleat/Tiling-Assistant/pull/171)

Unfortunately, trying to add gaps to titlebar-double-clicked windows has 2 big issues (see the PR)

- double-clicking to unmaxize doesn't really work
- animations aren't working

These issues are *veeery* noticable. So with these limitations I'd rather not add this feature since there are alternative behaviours (dragging window to the top or keyboard shortcuts).

**Comment by o-kotb** [comment]
For the last one, how about having the extension only work for corners? That way edges can be handled by GNOME and has the tiled style like with maximizing. I can then also disable the left and right edge extension keybinds to have it on GNOME's side as well. I think this is better than no corner tiling at all.

**Comment by Leleat** [comment]
> For the last one, how about having the extension only work for corners? That way edges can be handled by GNOME and has the tiled style like with maximizing. I can then also disable the left and right edge extension keybinds to have it on GNOME's side as well. I think this is better than no corner tiling at all.

I don't think that will work because GNOME has some basic tiling features like resizing side-by-side windows automatically. I won't be able to change those however because they aren't handled gnome-shell. That would break a lot of Tiling Assistant's features.

**Comment by 0Raik** [comment]
Incompatible with the extension

https://extensions.gnome.org/extension/1181/maximize-to-workspace/

Maximize to workspace is suppose to move a newly created/maximized window automatically to a new workspace which makes working with workspaces a breeze. So one can only choose that or Tiling Assistant and pretty gaps.

**Comment by rungitringit** [comment]
Hello,
If I was to request quarter tiling upstream (that is pin to corners and shortcut keys such as WIN + DOWN RIGHT) which project would be appropriate?  Mutter or Gnome?
Thank you!

**Comment by Leleat** [comment]
> If I was to request quarter tiling upstream (that is pin to corners and shortcut keys such as WIN + DOWN RIGHT) which project would be appropriate? Mutter or Gnome?
> Thank you!

Hi, the project to request this from would be mutter (that's why there are a limitations when it comes to tiling via gnome shell extensions). But you don't need to request it there. Better tiling has been requested a few times already (see links below). It's just no one has taken to implement it yet because (among other things) it's also lacking a proper design. The design team has been thinking about better tiling in GNOME but they are still in an explorative stage: https://blogs.gnome.org/tbernard/2023/07/26/rethinking-window-management/

---

https://gitlab.gnome.org/GNOME/mutter/-/issues/2128
https://gitlab.gnome.org/GNOME/mutter/-/issues/675
https://gitlab.gnome.org/GNOME/mutter/-/issues/52
https://gitlab.gnome.org/GNOME/mutter/-/issues/1596
https://gitlab.gnome.org/GNOME/mutter/-/issues/966
https://gitlab.gnome.org/GNOME/mutter/-/issues/704

**Comment by rungitringit** [comment]
Thank you for that.

> It's just no one has taken to implement it yet because (among other things) it's also lacking a proper design. The design team has been thinking about better tiling in GNOME but they are still in an explorative stage: https://blogs.gnome.org/tbernard/2023/07/26/rethinking-window-management/

Oh no, that post scares me.  I like that they're trying to think outside the box to improve the design overall, but I am worried that they'll come up with some radical idea which will take years to mature.  I often need to use "mature" OS releases at work so the mistakes of the past have a long tail!

If they could simply match MS Windows tiling configuration today, it would be a huge step forward for most users.

**Comment by rungitringit** [comment]
FWIW I've used work contacts to raise this as a feature request with Red Hat.  That's got to be worth more than another :+1: on a gitlab.gnome.org Issue.

**Comment by Fenimoure** [comment]
Thank you, wanted to report windows that open maximized by default not having a gap and found it's not your issue

**Comment by Fenimoure** [comment]
> maximized by default not having a gap

Also, I think there is a workaround possible: for such windows automatically de-maximize them and then re-maximize them for having a gap. I also think, if implemented, this should be optional, because it may irritate some users.
